shell/store/type-map.js

```diff
@@ -217,7 +217,9 @@ export function DSL(store, product, module = 'type-map') {
     headers(type, headers) {
       headers.forEach((header) => {
         // If on the client, then use the value getter if there is one
-        if (header.getValue) {
+        if (process.client && header.getValue) {
+          // we need to store the .value prop for the advanced filtering
+          header.valueProp = header.value;
```

Comment:
We need to use the `value` prop (which in case of `header.getValue` get overwritten hence why we are storing it on another prop) so that we can filter by the correct prop. 
`header.getValue` is a function which I cannot use to get the prop I need, afaik. Also `header.value` is present in 99% percent of the cases in table header definitions, so that's an important plus.

additionally we also filter by the `header.sort` combining them all into an array of properties to filter by. Tbh, I don't remember why I added them as "filters" to a given col, but if I had to guess it was to cover some edge-case on which we didn't have the `header.value` defined but we had the `header.sort` with the correct information there to filter by.

Reply:
Alex and I chatted about this and we both understand more.

Underlying this was a misalignment on how getValue works 
- getValue should be seen as a replacement for value
- getValue receives the row and returns an arbitrary value that's not necessarily linked to the property name in `value`

Before applying this outside of elemental we'll need to address this. At the moment advance filtering won't work on columns where only getValue is used, or getValue differs from value (which in itself is a miss-configuration)